pkg/validator/app.go

```diff
@@ -151,19 +152,26 @@ func validateSelectedRoles(
 // validateRoles checks each role for property constraints not expressible
 // in the schema. If any overrideable properties are unspecified, the corresponding
 // global values are used. This will add an PATCH spec for mutation the app CR.
+// The role in appCR will be correspondingly updated so that it can later be
+// to check whether the resulting CR differs from the current stored appCR.
 func validateRoles(
 	appCR *kdv1.KubeDirectorApp,
 	patches []appPatchSpec,
 	valErrors []string,
 ) ([]appPatchSpec, []string) {
 
+	// Any global defaults will be removed from the CR. Remember their values
+	// though for use in populating the role definitions.
 	var globalImageRepoTag *string
 	var globalSetupPackageURL *string
 	var globalPersistDirs *[]string
 	var globalEventList *[]string
```

Comment:
I would leave a blank line here for more readability but that is how I like to read my code :)

Reply:
globalDefaults here means common across all roles, is it?

Reply:
yep